protos/flyteidl/admin/workflow.proto

```diff
@@ -65,3 +65,21 @@ message WorkflowClosure {
     // Time at which the workflow was created.
     google.protobuf.Timestamp created_at           = 2;
 }
+
+// The workflow id is already used and the structure is different
+message WorkflowErrorExistsDifferentStructure {
+    core.Identifier id = 1;
+}
+
+// The workflow id is already used with an identical sctructure
+message WorkflowErrorExistsIdenticalStructure {
+    core.Identifier id = 1;
+}
+
+// When a CreateWorkflowRequest failes due to matching id
+message CreateWorkflowFailureReason {
```

Comment:
Should we do the same for tasks?

Reply:
What if we do that in a seperate issue/ pr series? I can make the issue for it as a follow up after finishing this one if you want.

Reply:
That would be awesome!